FLT/Mathlib/Topology/Algebra/Group/Units.lean

```diff
@@ -7,9 +7,88 @@ lemma Submonoid.units_isOpen {M : Type*} [TopologicalSpace M] [Monoid M]
   {U : Submonoid M} (hU : IsOpen (U : Set M)) : IsOpen (U.units : Set Mˣ) :=
   (hU.preimage Units.continuous_val).inter (hU.preimage Units.continuous_coe_inv)
 
+ -- FLT#588
 lemma Submonoid.units_isCompact {M : Type*} [TopologicalSpace M] [Monoid M] [ContinuousMul M]
     [T2Space M] {U : Submonoid M} (hU : IsCompact (U : Set M)) : IsCompact (U.units : Set Mˣ) := by
-  sorry -- FLT#588
+  let φ : Mˣ → M × M := fun x ↦  (↑x, ↑x⁻¹)
+  let p : M × M → M := fun (u,v) => u * v
+  let q : M × M → M := fun (u,v) => v * u
```

Comment:
In Lean's type theory, deconstructing inputs like this gives you functions which contain `match`es and are generally a bit more annoying to work with. Probably
```
  let p : M × M → M := fun uv => uv.1 * uv.2
  let q : M × M → M := fun uv => uv.2 * uv.1
```
is better; this is mathematically the same but probably easier to work with in practice.